articles/140_resource_names.md

```diff
@@ -38,152 +58,157 @@ A valid name has the following characteristics:
 Exception: base names (described below) cannot have forward slashes (/) or tildes (~) in them.
 ```
 
-### DDS Topic Names
+#### DDS Topic Names
 
-In DDS, topic names are restricted by these restrictions:
+In DDS, topic names restrictions include:
 
-*From DDS 1.4 specification, or the [RTI documentation](http://community.rti.com/rti-doc/510/ndds/doc/html/api_cpp/group__DDSQueryAndFilterSyntaxModule.html):*
+*From DDS 1.4 specification, or [RTI documentation](http://community.rti.com/rti-doc/510/ndds/doc/html/api_cpp/group__DDSQueryAndFilterSyntaxModule.html):*
 
 ```
 TOPICNAME - A topic name is an identifier for a topic, and is defined as any series of characters 'a', ..., 'z', 'A', ..., 'Z', '0', ..., '9', '_' but may not start with a digit.
 ```
 
-*Note:* that the DDS specification has a known typo, where it says `-` are allowed, but the RTI documentation correctly lists `_` as allowed.
+*Note:* the DDS specification has a known typo, where it says `-` are allowed, but the RTI documentation correctly lists `_` as allowed.
 
 Additionally, DDS - or more precisely the underlaying RTPS protocol - has a hard limit on topic names of 256 characters, so an additional goal is to minimize the number of extra characters used when mapping from ROS to DDS names.
 See The Real-time Publish-Subscribe Protocol (RTPS) DDS Interoperability Wire Protocol Specification, Table 9.12 for more details.
 
-## ROS 2 Topic and Service Name Constraints
+## Uniform Resource Locators (URLs)
 
-In this section an outline of the proposed constrains for ROS 2 topic and service names will be enumerated along with rationales where appropriate.
+Resources are addressed by means of the [Uniform Resource Locators (URL)](https://en.wikipedia.org/wiki/Uniform_Resource_Locator) format.
 
-For convenience here is a summary of all rules for topic and service names in ROS 2:
+### Syntax Components
 
-- must not be empty
-- may contain alphanumeric characters (`[0-9|a-z|A-Z]`), underscores (`_`), or forward slashes (`/`)
-- may use balanced curly braces (`{}`) for substitutions
-- may start with a tilde (`~`), the private namespace substitution character
-- must not start with a numeric character (`[0-9]`)
-- must not end with a forward slash (`/`)
-- must not contain any number of repeated forward slashes (`/`)
-- must not contain any number of repeated underscores (`_`)
-- must separate a tilde (`~`) from the rest of the name with a forward slash (`/`), i.e. `~/foo` not `~foo`
-- must have balanced curly braces (`{}`) when used, i.e. `{sub}/foo` but not `{sub/foo` nor `/foo}`
+#### Provider Names
```

Comment:
So, the rationale here is that since nodes provide services instead of hosting data it made more sense to use the former terminology, but the distinction is subtle. I'm fine with `host` too. See 7680627ee8f2fa8675fd58e72f814585c6721f2a.

Reply:
Yeah, I don't feel strongly about it, host is fine and I would be ok with provider as well so long as there is a rationale and it's not arbitrary.